presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -721,7 +721,7 @@ public static long levenshteinDistance(@SqlType("varchar(x)") Slice left, @SqlTy
     @SqlType(StandardTypes.BIGINT)
     public static long hammingDistance(@SqlType("varchar(x)") Slice left, @SqlType("varchar(y)") Slice right)
     {
-        int distance = 0;
+        long distance = 0;
```

Comment:
yup, I suggest using ```Int``` as a return type, hence limit the string size. Otherwise, calculation until ```Long.MaxValue``` halt worker for a long time.